src/spatialdata/_core/query/spatial_query.py

```diff
@@ -747,7 +758,12 @@ def polygon_query(
         vv = pd.concat(v)
         vv = vv[~vv.index.duplicated(keep="first")]
         geodataframes[k] = vv
+    if filter_table:
+        elements = {"shapes": geodataframes}
+        tables: dict[str, AnnData] | Tables = {
+            name: _filter_table_by_elements(table, elements) for name, table in sdata.tables.items()
+        }
+    else:
+        tables = sdata.tables
```

Comment:
this code block appears 3 times, I'd put it into a tiny private function

Reply:
Resolved with [0aa05fe](https://github.com/scverse/spatialdata/pull/410/commits/0aa05fe65ea5c4ef03d2dd25d15e00f65686b791)